plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -811,6 +811,10 @@ def save_fileless_units(self, file_handle, tag, process_func, mutable_type=False
                 (model for model in all_packages if model.unit_key_as_named_tuple in to_save)
 
         for model in package_info_generator:
+            # Add repo_id to each collection of the pkglist of the new erratum
+            if isinstance(model, models.Errata):
+                for collection in model.pkglist:
+                    collection['_pulp_repo_id'] = self.repo.repo_id
```

Comment:
I understand now why the `_pulp_repo_id` is being stored. It makes sense. Thanks for the explanation.